### PR TITLE
Add e2e test to validate that hostProcess containers can use inclusterclient

### PR DESCRIPTION
**Pull request by marosset**
Signed-off-by: Mark Rossetti <marosset@microsoft.com>

#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

Adds an e2e test case that uses `agnhost inclusterclient` from a hostProcess container to verify in-cluster config / client works in them.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
[KEP]: https://github.com/kubernetes/enhancements/issues/1981
```

/sig windows
/milestone v1.26

### CODE REVIEW

**Comment by jsturtevant** [comment]
related to https://github.com/kubernetes/kubernetes/issues/104562

File: test/e2e/windows/host_process.go [hunk 1]
```diff
@@ -600,6 +600,82 @@ var _ = SIGDescribe("[Feature:WindowsHostProcessContainers] [MinimumKubeletVersi
...
+						RunAsUserName: &User_NTAuthoritySystem,
+					},
+				},
+				HostNetwork: true,
+				Containers: []v1.Container{
+					{
+						Image: imageutils.GetE2EImage(imageutils.Agnhost),
+						Name:  "hpc-agnhost",
+						// TODO: figure out why we need to copy agnhost as agnhost.exe here
+						Command: []string{"cmd", "/C", "copy", "c:\\hpc\\agnhost", "c:\\hpc\\agnhost.exe", "&&", "c:\\hpc\\agnhost.exe", "inclusterclient"},
```
Comment (jsturtevant) [inline test/e2e/windows/host_process.go:640]: thought this was fixed in https://github.com/microsoft/hcsshim/issues/1128, Maybe related new functionality in HPC that enables the volume projection?

File: test/e2e/windows/host_process.go [hunk 2]
```diff
@@ -600,6 +600,82 @@ var _ = SIGDescribe("[Feature:WindowsHostProcessContainers] [MinimumKubeletVersi
...
+				},
+			},
+		}
+
+		pc := e2epod.NewPodClient(f)
+		pc.Create(pod)
+
+		// We wait an additional 60 seconds after the pod is Running because the
+		// `InClusterClient` app in the agnhost image waits 30 seconds before
+		// starting to make queries - https://github.com/kubernetes/kubernetes/blob/ceee3783ed963497db669504241af2ca6a11610f/test/images/agnhost/inclusterclient/main.go#L51
```
Comment (jsturtevant) [inline test/e2e/windows/host_process.go:655]: nit: seems like comment in on wrong line :-)

File: test/e2e/windows/host_process.go [hunk 3]
```diff
@@ -600,6 +600,82 @@ var _ = SIGDescribe("[Feature:WindowsHostProcessContainers] [MinimumKubeletVersi
...
+
+		framework.ExpectEqual(
+			strings.Contains(logs, "calling /healthz"),
+			true,
+			"app logs should contain 'calling /healthz'")
+
+		framework.ExpectEqual(
+			strings.Contains(logs, "status=failed"),
+			false,
+			"app logs should not contain 'statued=failed")
```
Comment (jsturtevant) [inline test/e2e/windows/host_process.go:676]:
```suggestion
			"app logs should not contain 'status=failed")
```

File: test/e2e/windows/host_process.go (hunk 1 above)
Comment (marosset) [reply test/e2e/windows/host_process.go:640]: Hmm, I'll take a look and as a minimum leave update the comment here.

Comment (marosset) [reply test/e2e/windows/host_process.go:643]: I updated the comment for now.

**Comment by marosset** [comment]
will rebase soon

**Pull request merged**